provider.go

```diff
@@ -21,9 +27,10 @@ package g8
 //     gate := g8.NewGate(g8.NewAuthorizationService().WithClientProvider(clientProvider))
 //
 type ClientProvider struct {
-	cache bool
-
+	cache                bool
 	getClientByTokenFunc func(token string) *Client
+	gocache              *gocache.Cache
```

Comment:
Could you rename this to `cache` and remove the existing `cache` field? We can just check if `cache == nil`, to know if the cache is enabled or not.

Reply:
Yes absolutely!